0023-backend-v3.md

```diff
@@ -0,0 +1,269 @@
+```
+
+Where `Target + .run` are **required** components, and `"extra functionality"` comprises both optional and custom functionality that can be used to extend the backend model and adapt it using provider-specific information. 
+
+The initial proposal in this RFC is to **maintain** the `BackendV2` design of the Target and extra functionality (up to small changes), and focus on improving the "run" component. The detailed design section expands on what each component involves.
+
+### BackendV3 - the new backend model
+
+Instead of a required `backend.run()`, the proposal for `BackendV3` implementations is to offer **primitive factory methods**: `backend.sampler()` and `backend.estimator()` that return their respective instantiated primitive classes. These factory methods should be defined in the abstract `BackendV3` interface and it's up to the provider to decide what "flavor" of primitive will be returned in their specific `BackendV3` subclass. 
+
+The abstract definition of factory methods should be: 
+
+- as minimal as possible without being trivial
+- agnostic to the primitive interface version (to support both current and future versions of the primitives)
+
+We could additionally still maintain an optional `backend.run()` for 3rd party and local simulation backends.
+
+This sums up the change from having one mandatory implementation of the execution step (`.run()`) to having three optional implementations: 
+
+- `.sampler()`
+- `.estimator()`
+- `.run()`
+
+Important design questions include:
+
+1. **Required vs optional**
+
+`BackendV2` only offered 1 required execution method `.run()`. With `BackendV3`, there are 3 paths for execution: should we make them all optional and not require backends to implement the "run" component? Should we require at least a `.sampler()` or a `.estimator()` method (which return a primitive and don't actually execute)?
+
+Making all 3 methods optional seem like the best choice because of its flexibility: it enables all three execution options as well as an "execution-less" backend object that only exposes a model for compilation (transpilation). This can be different from just using a `Target` because of the additional functionality backend implementations provide.
+
+2. **The role of .run, why is it valuable** 
+
+The role of `run()`, as defined in `BackendV2` often gets immediately translated to "return counts", which is more restrictive than what the interface dictates. `run()` is designed to be free form, the only requirements are that the input is  a `QuantumCircuit` and the output is a `Result`. `Result` is purposely unrestricted, there are no requirements on the data it contains and the implementation details are left up to the implementer/vendor. There is a bias/convention towards returning `Counts` per circuit input in the job for historical reasons, and there are some formatting expectations if using those count mechanisms. But the output of `.run()` doesn't have to be "counts". 
```

Comment:
I have trouble finding the value of abstracting an execution mode whose implementation is optional, and when implemented, whose output is free-form. It seems like the real motivation for keeping .run in the abstraction is for historical continuity.